docs/react-types/ReactNode.md

```diff
@@ -0,0 +1,49 @@
+---
+title: ReactNode
+---
+
+`ReactNode` is a type that describes what React can render.
+
+## Parameters
+
+`ReactNode` does not take any parameters.
+
+## Usage
+
+### Typing `children`
+
+The most common use case for `ReactNode` is typing `children`.
+
+```tsx
+interface Props {
+  children?: ReactNode;
+}
+
+function Component({ children }: Props) {
+  return <>{children}</>;
+}
+```
+
+`<Component>` accepts anything that React can render as `children`. Here are some examples:
+
+```tsx
+function Examples() {
+  return (
+    <>
+      <Component>
+        <div>Hello</div>
+      </Component>
+      <Component>Hello</Component>
+      <Component>{123}</Component>
+      <Component>
+        <>Hello</>
+      </Component>
+      <Component>{createPortal(<>Hello</>, document.body)}</Component>
```

Comment:
It should be obvious where this comes from. We should probably have this separated since it's DOM specific at the moment.

Reply:
That's true, removed it for now since this is only a list of some example – doesn't need to be exhaustive.